src/Controls/src/Core/Platform/iOS/ModalWrapper.cs

```diff
@@ -34,7 +34,7 @@ internal ModalWrapper(IPlatformViewHandler modal)
 				{
 					Color modalBkgndColor = ((Page)_modal.VirtualView).BackgroundColor;
 
-					if (modalBkgndColor?.Alpha > 0)
+					if (modalBkgndColor?.Alpha > 0 || modalBkgndColor == Colors.Transparent)
```

Comment:
@jsuarezruiz for this check here can we do this instead? 

```C#
if (modalBkgndColor?.Alpha < 1)
```

I see that this conditional first made it in on this PR. 

https://github.com/xamarin/Xamarin.Forms/pull/11915

Reply:
but wasn't this merged already to main?

Reply:
https://github.com/dotnet/maui/issues/12031

Reply:
Unless I'm thinking of it wrong

```C#
if (modalBkgndColor?.Alpha > 0 || modalBkgndColor == Colors.Transparent)
```

Will be true 100 percent of the time, right?

What's a condition where the above if statement is false? 

The point of the if statement is to change `FullScreen` to `OverFullScreen` if the user wants to see the underlying view still. This behavior broke in forms with this [PR](https://github.com/xamarin/Xamarin.Forms/pull/11915) because

```C#
if (modalBkgndColor?.Alpha > 0)
```

Means if the modal background is full transparent then it will still use `FullScreen`

It seems like the if statement should be 

```C#
if (modalBkgndColor?.Alpha < 1)
```

Which says if the modal is opaque then don't do anything but if it has any transparency then do something

Reply:
Also, this code is only checking `BackgroundColor` so if the developer is using `Background` then AFAICT this code is still broken.

Reply:
That being said, I don't think this code makes anything more wrong. 

It makes it fractionally more correct so it's probably fine to backport, but anyone using "Background" vs "BackgroundColor" will still be broken.

Reply:
We can merge https://github.com/dotnet/maui/pull/12134 and close this one. Then later, backport 12134.